### ISSUE TITLE
How can I have different "Edit this page" URL for each page

### ISSUE THREAD

**Issue by undergroundwires** [alex-shpak/hugo-book#150]
**Background**
- I have two repositories consisting of ".md" files.
- I created a script that clones them, remaps file structure (see #149) and adds `hugo-book` metadata. In the end the script creates ready-to-go `content/` and `static/` folders.

**What I'm trying to achieve**
- Edit this page URL for each page.

**Problem**
- Master files exist in two different repositories.
- Folder & file name structure in those repositories does not match the file structure of the website.
- So I can't use `BookRepo` and `BookEditPath`

**Comment by alex-shpak** [comment]
Hi!
I would say there is no easy solution.

What you can try:
 1. Add url (or path) of original file to frontmatter of each `.md` file. For example:
```md
---
BookEditPath: your/path/here
---
```

 2. Overwrite `/layouts/partials/docs/footer.html`. You don't need to fork theme, you can override that specific file in your website. And make use of `.Params.BookEditPath` instead of `.Site.Params.BookEditPath` and change [this line](https://github.com/alex-shpak/hugo-book/blob/master/layouts/partials/docs/footer.html#L18) accordingly 

And then if you plan to migrate fully to hugo - remove this change after.

**Comment by alex-shpak** [comment]
Also.

I thought for some time to add [Nested menu](https://gohugo.io/content-management/menus/#nesting) support. Maybe that would solve your problem.

There are 2 menus: `menu: before` and `menu: after` in theme. they are native hugo menus and page can be attached to it via frontmatter.

[About](https://gohugo.io/content-management/menus/)

**Comment by alex-shpak** [comment]
So I just added it to master. But then realised that it would not solve your problem with collapsing menu 😞 

So overwriting template I guess is the option.

**Comment by undergroundwires** [comment]
@alex-shpak thanks for your fast responses and supporting this project. It's awesome that you do that.

Menu nesting is not a direct solution but it's cool that hugo-book supports it now 👍 

I really don't want to patch my local copy of the theme. I simply build another customized site generator on top of hugo and hugo-book where I iterate each markdown file and add edit link on them.

I can try to look at it & send a PR if you think hugo-book should natively support something like `PageEditUrl` that overrides `BookEditPath` and `BookRepo` if it has any value.

**Comment by alex-shpak** [comment]
Well, you don't need patch theme itself but overwrite template with Hugo.
Something like:

```
yoursite/
  themes/book/... <- theme files
  layouts/partials/docs/footer.html <- your custom changes
```

In this case you need only to watch theme changes for that specific file.
A bit ugly idea: if you edit files automatically you can disable theme's edit button and generate link to edit in the end of the file.

As for adding `PageEditUrl` I think it is quite specific case, so I don't want to add it to theme, you know, keep it simple. Sorry.

**Comment by alex-shpak** [comment]
There are also `inject` templates that you can use as hooks
https://github.com/alex-shpak/hugo-book/tree/master/layouts/partials/docs/inject
https://github.com/alex-shpak/hugo-book#partials

**Comment by undergroundwires** [comment]
Thanks for all explanation, I learnt a lot today.

I loved the solution with inject templates 👌. It seems to be the cleanest way, and makes the theme very extensible 💪.

I decided to have the buttons on the top, right before the page content. I see that there's an `inject` template for `footer.html` but not `page-header.html` or something similar. I tried to override `header.html` but I see that it's a mobile-only partial.

I got it working by adding `shortcodes` in markdown files using a script, a dirty workaround but does the work, and I don't need to watch for the theme changes =)

![image](https://user-images.githubusercontent.com/15555035/75101527-16b0af00-55de-11ea-8558-1e7ce611bb69.png)
